src/decoder_fmmidi.cpp

```diff
@@ -57,9 +57,9 @@ bool FmMidiDecoder::Open(FILE* file) {
 	fseek(file, 0, SEEK_END);
 	file_buffer.resize(ftell(file) - old_pos);
 	fseek(file, old_pos, SEEK_SET);
-	fread(file_buffer.data(), file_buffer.size(), 1, file);
+	size_t bytes_read = fread(file_buffer.data(), file_buffer.size(), 1, file);
 
-	if (!seq->load(this, read_func)) {
+	if ((bytes_read =! file_buffer.size()) || (!seq->load(this, read_func))) {
```

Comment:
the operator should be != . I guess this assigns 0 or 1 (negation of file_buffer.size() to bytes_read.